core/src/main/java/jenkins/model/IComputer.java

```diff
@@ -0,0 +1,180 @@
+
+/**
+ * Interface for computer-like objects meant to be passed to {@code t:executors} tag.
+ *
+ * @since TODO
+ */
+@Restricted(Beta.class)
+public interface IComputer extends AccessControlled {
+    /**
+     * Returns {@link Node#getNodeName() the name of the node}.
+     */
+    @NonNull
+    String getName();
+
+    /**
+     * Used to render the list of executors.
+     * @return a snapshot of the executor display information
+     */
+    @NonNull
+    List<? extends IDisplayExecutor> getDisplayExecutors();
+
+    /**
+     * @return {@code true} if the node is offline. {@code false} if it is online.
+     */
+    boolean isOffline();
+
+    /**
+     * @return the node name for UI purposes.
+     */
+    @NonNull
+    String getDisplayName();
+
+    /**
+     * Returns {@code true} if the computer is accepting tasks. Needed to allow agents programmatic suspension of task
+     * scheduling that does not overlap with being offline.
+     *
+     * @return {@code true} if the computer is accepting tasks
+     * @see hudson.slaves.RetentionStrategy#isAcceptingTasks(Computer)
+     * @see hudson.model.Node#isAcceptingTasks()
+     */
+    boolean isAcceptingTasks();
+
+    /**
+     * @return the URL where to reach specifically this computer, relative to Jenkins URL.
+     */
+    @NonNull
+    String getUrl();
+
+    /**
+     * @return {@code true} if this computer has a defined offline cause, @{code false} otherwise.
+     */
+    default boolean hasOfflineCause() {
+        return getOfflineCauseReason() != null;
+    }
+
+    /**
+     * If the computer was offline (either temporarily or not),
+     * this method will return the cause as a string (without user info).
+     * <p>
+     * {@code hasOfflineCause() == true} implies this must be non-null.
+     *
+     * @return
+     *      empty string if the system was put offline without given a cause.
+     */
+    @Nullable
+    String getOfflineCauseReason();
+
+    /**
+     * @return true if the node is currently connecting to the Jenkins controller.
+     */
+    boolean isConnecting();
+
+    /**
+     * Returns the icon for this computer.
+     * <p>
+     * It is both the recommended and default implementation to serve different icons based on {@link #isOffline}
+     *
+     * @see #getIconClassName()
+     */
+    String getIcon();
+
+    /**
+     * Returns the alternative text for the computer icon.
+     */
+    String getIconAltText();
+
+    /**
+     * Returns the class name that will be used to look up the icon.
+     * <p>
+     * This class name will be added as a class tag to the html img tags where the icon should
+     * show up followed by a size specifier given by {@link Icon#toNormalizedIconSizeClass(String)}
+     * The conversion of class tag to src tag is registered through {@link IconSet#addIcon(Icon)}
+     *
+     * It is both the recommended and default implementation to serve different icons based on {@link #isOffline}
```

Comment:
There is no _default implementation_. If there is one in `Computer`, that should be noted there, with `@inheritDoc`.